src/app/view/components/dot-device-selector/dot-device-selector.component.ts

```diff
@@ -51,4 +60,9 @@ export class DotDeviceSelectorComponent implements OnInit {
             ...devices
         ];
     }
+
+    private setDropdownWidth(devices: DotDevice[]): void {
+        const optionValues = devices.map((deviceOption: DotDevice) => deviceOption.name);
+        this.dropdownWidth = StringPixels.getWidth(optionValues) + this.arrowDropdownComponentSize;
```

Comment:
We should have a `getDropdownWidth` in the `StringPixels` and that way we can put the `arrowDropdownComponentSize` in one specific place, at the end of the day the dropdown is why we're doing all this and should also return the value as string with the `+ px`.